app/models/affair.rb

```diff
@@ -240,9 +250,19 @@ def status
   # Reload and save! affair to force counts and sums computation
   def refresh!
     reload
+    update_third
     save!
   end
 
+  def update_third
+    entities_ids = []
+    self.class.affairable_types_entity_foreign_keys.each do |model, foreign_key|
+      entities_ids << model.to_s.camelize.constantize.where(affair_id: id).pluck(foreign_key)
```

Comment:
Polymorphism would make this something like `model.where(affair: self).pluck(:third_id)`
(Maybe we need to look into attribute aliasing, I'm pretty sure ARecord can do that.